app/src/androidTest/java/com/willowtree/vocable/room/MigrationTest.kt

```diff
@@ -61,4 +64,117 @@ class MigrationTest {
+        val creationDateOne = System.currentTimeMillis()
+
+        val phraseTwoId = UUID.randomUUID().toString()
+        val creationDateTwo = System.currentTimeMillis()
+
+        helper.createDatabase(TEST_DB, 3).apply {
+            // Create mock My Sayings category
+            execSQL("INSERT INTO Category (category_id, creation_date, is_user_generated, localized_name, hidden, sort_order) VALUES ('${PresetCategories.USER_FAVORITES.id}', ${System.currentTimeMillis()}, 0, '$mySayingsCategoryNameV3', 0, ${PresetCategories.USER_FAVORITES.initialSortOrder})")
+
+            // Add custom phrases to My Sayings category
+            execSQL("INSERT INTO Phrase (phrase_id, creation_date, is_user_generated, last_spoken_date, localized_utterance, sort_order) VALUES ('$phraseOneId', $creationDateOne, 1, $creationDateOne, '$mySayingTestPhraseOneV3', 0)")
+            execSQL("INSERT INTO CategoryPhraseCrossRef (category_id, phrase_id) VALUES ('${PresetCategories.USER_FAVORITES.id}', '$phraseOneId')")
+
+            execSQL("INSERT INTO Phrase (phrase_id, creation_date, is_user_generated, last_spoken_date, localized_utterance, sort_order) VALUES ('$phraseTwoId', $creationDateTwo, 1, $creationDateTwo, '$mySayingTestPhraseTwoV3', 1)")
+            execSQL("INSERT INTO CategoryPhraseCrossRef (category_id, phrase_id) VALUES ('${PresetCategories.USER_FAVORITES.id}', '$phraseTwoId')")
+            close()
+        }
+
+        helper.runMigrationsAndValidate(TEST_DB, 4, true, VocableDatabaseMigrations.MIGRATION_3_4)
+            .apply {
+                // Verify that new schema is as expected
+                val categoryId = PresetCategories.USER_FAVORITES.id
+                val crossRefCursor =
+                    query("SELECT phrase_id FROM CategoryPhraseCrossRef WHERE category_id = '$categoryId'")
+                val myLocalizedSayings = arrayListOf<String>()
+                val phraseIds =  mutableListOf<String>()
+                while (crossRefCursor.moveToNext()) {
+                    val phraseId = crossRefCursor.getString(crossRefCursor.getColumnIndex("phrase_id"))
+                    phraseIds.add(phraseId)
+                }
+                crossRefCursor.close()
+
+                phraseIds.forEach {
+                    val phraseCursor = query("SELECT localized_utterance FROM Phrase WHERE phrase_id = '$it'")
+                    while (phraseCursor.moveToNext()) {
+                        val saying = phraseCursor.getString(phraseCursor.getColumnIndex("localized_utterance"))
+                        myLocalizedSayings.add(saying)
+                    }
+                    phraseCursor.close()
+                }
+                Assert.assertTrue(myLocalizedSayings.contains(mySayingTestPhraseOneV3))
+                Assert.assertTrue(myLocalizedSayings.contains(mySayingTestPhraseTwoV3))
+                close()
+            }
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun migrate3to4_with_legacySayings() {
```

Comment:
This should probably use camel case instead of snake case 🐫

Reply:
lol what about this hybrid